### DISCUSSION TITLE
10 luglio 2023 - esercizio 2

### DISCUSSION THREAD

**Discussion by IonutCicio** [sapienzastudentsnetwork/basi-di-dati-1#42] [Esami]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/40034429/c8063e4f-a25f-46a0-a47a-e48b794c55c2)

**Comment by Warcophyr** [comment]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/18f28e94-6d68-4eed-a605-c475e13ba6a8)
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/ed6403e2-df3d-491d-ab8f-17b836adabd9)
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/2a1bf79a-39db-436f-8bd8-6e3e9bb35ac9)
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/870a4af9-f1ee-4826-aa35-cf71595c4abb)
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/a3c3f947-a563-489e-81be-d178d74b8261)

**Reply by Alby7503** [reply]
Come hai scelto nell’ultimo insieme di aggiungere BFGA piuttosto che BFGC o BFGD?

**Reply by atyion** [reply]
Basta che scegli una qualsiasi chiave.

**Comment by atyion** [comment]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/101365532/56638d97-826b-4837-a858-cdf386347bea)

**Comment by niccolo-fato** [comment]
![IMAGE 2024-01-14 20:27:12](https://github.com/sapienzastudentsnetwork/bdi2324/assets/73995334/a91acb3f-a8c4-424d-8400-6fc3dce70490)
![IMAGE 2024-01-14 20:27:19](https://github.com/sapienzastudentsnetwork/bdi2324/assets/73995334/c429ba79-aad1-4321-bf48-eed3d1183507)

**Comment by Frisbon** [comment]
### Dettaglio importante nell'algoritmo di decomposizione

> Codice prestato da @CuriousCI:

```
def decomposition(R, F: minimal cover):
    S = ∅
    ρ = ∅

    for A ∈ R | ∄ X → Y ∈ F : A ∈ XY:
        S = S ∪ {A}

    if S != ∅:
        R = R - S
        ρ = ρ ∪ {S}

    if ∃ X → Y ∈ F | XY = R:
        ρ = ρ ∪ {R}
    else:
        for X → A ∈ F:
            ρ = ρ ∪ {XA}
```

- Nel primo for, S diventa prima {F} e poi {F,G} perché ∅ ∪ {F} ∪ {G} **non é** {FG}!

Difatti, ho fatto un errore nella decomposizione finale aggiungendo FG, al posto di F,G (separati)

PS: @niccolo-fato ti taggo per farti notare che hai avuto la stessa confusione

**Reply by niccolo-fato** [reply]
Sì Si, lo so. Questa cosa l'ho scoperta praticamente 10 minuti prima di fare l'esame scritto(infatti non lo faccio in nessuno esercizio). La forma corretta della decomposizione dovrebbe essere la seguente:
![2024-02-11 11 54 24](https://github.com/sapienzastudentsnetwork/bdi2324/assets/73995334/65bc6cf8-42c9-4028-af84-5dc92b5853a5)

**Comment by M1SC0** [comment]
<img width="1391" height="1022" alt="image" src="https://github.com/user-attachments/assets/d5b43f89-85d8-46f5-9416-4ee1a93329d7" />

<img width="896" height="978" alt="image" src="https://github.com/user-attachments/assets/8ccbc178-6ead-4864-a3d3-1bc6c48c69ab" />
